shiny/session/_session.py

```diff
@@ -692,7 +694,12 @@ def _manage_inputs(self, data: dict[str, object]) -> None:
             # The keys[0] value is already a fully namespaced id; make that explicit by
             # wrapping it in ResolvedId, otherwise self.input will throw an id
             # validation error.
-            self.input[ResolvedId(keys[0])]._set(val)
+            k = keys[0]
+            self.input[ResolvedId(k)]._set(val)
+
+            if k.startswith(".clientdata_"):
+                k2 = k.split("_", 1)[1]
+                self.clientdata[ResolvedId(k2)]._set(val)
```

Comment:
Could we include one example of `k` for a full client data input in the comments? Also would be worth linking to the upstream source where this contract originates